pkg/schemas/service.go

```diff
@@ -223,6 +227,59 @@ func (s *Service) DeleteSchema(ctx context.Context, ID string) error {
 
 }
 
+func (s *Service) GetDefaultSchema(ctx context.Context) (*DefaultSchema, error) {
+	ctx, span := s.tracer.Start(ctx, "schemas.Service.GetDefaultSchema")
+	defer span.End()
+
+	cm, err := s.k8s.ConfigMaps(s.cmNamespace).Get(ctx, s.cmName, metaV1.GetOptions{})
+
+	if err != nil {
+		s.logger.Error(err.Error())
+		return nil, err
+	}
+
+	ID, ok := cm.Data[DEFAULT_SCHEMA]
+
+	if !ok {
+		return nil, fmt.Errorf("default schema %s missing", DEFAULT_SCHEMA)
+	}
+
+	defaultSchema := new(DefaultSchema)
+	defaultSchema.ID = ID
+
+	return defaultSchema, nil
+}
+
+func (s *Service) UpdateDefaultSchema(ctx context.Context, schema *DefaultSchema) (*DefaultSchema, error) {
+	ctx, span := s.tracer.Start(ctx, "schemas.Service.UpdateDefaultSchema")
+	defer span.End()
+
+	cm, err := s.k8s.ConfigMaps(s.cmNamespace).Get(ctx, s.cmName, metaV1.GetOptions{})
+
+	if err != nil {
+		s.logger.Error(err.Error())
+		return nil, err
+	}
+
+	if _, ok := cm.Data[DEFAULT_SCHEMA]; !ok {
+		return nil, fmt.Errorf("default schema %s missing", DEFAULT_SCHEMA)
+	}
```

Comment:
I still don't understand why this is needed. If the default schema is missing, that means that the config is wrong. We would want to fix the broken config rather than throw an error.

Reply:
For me it wasn't obvious that a bad config shouldn't throw an error, but I accept your reasoning. fixed.